lib/internal/modules/run_main.js

```diff
@@ -81,6 +81,12 @@ function shouldUseESMLoader(mainPath) {
   if (mainPath && StringPrototypeEndsWith(mainPath, '.mjs')) { return true; }
   if (!mainPath || StringPrototypeEndsWith(mainPath, '.cjs')) { return false; }
 
+  if (getOptionValue('--experimental-strip-types')) {
+    // This will likely change in the future to start with commonjs loader by default
+    if (mainPath && StringPrototypeEndsWith(mainPath, '.ts')) { return true; }
```

Comment:
Currently in this PR I removed support for CJS because I could not make it work properly, so I guess for now its fine

Reply:
I could help out with the CJS changes but I think I prefer to see some resolution with the TypeScript team about whether this is actually what we are going for before jumping in.

Reply:
I think `.ts` files need to be interpreted the same as `.js` files, using the same full algorithm (`package.json` `type` field, `--experimental-default-type`, `--experimental-detect-module`, `--input-type`, etc.). If the result of the format determination is CommonJS, it can just error for now; that’s fine for this initial PR. But I don’t want us to determine one format for `.ts` files now and have that determination change in a follow-up. I consider this blocking.

Also the docs currently assume that CommonJS will work, as that was the case when they were written, so if we’re going to ship an ESM-only first version then the docs need updating.

Reply:
added support for everything expect `--input-type` and added lots of tests